bluesky/run_engine.py

```diff
@@ -270,7 +270,7 @@ def _clear_run_cache(self):
         self._metadata_per_run.clear()
         self._bundling = False
         self._run_is_open = False
-        self._msg_cache = None  # checkpoints can't rewind into a closed run
```

Comment:
We *do* want to support checkpoints between runs during a multi-run plan. ("If interrupted redo the last 5 samples....") It's better to handle the rewinding issue in `_close_run` itself.